ui/admin/app/components/model/DefaultModelAliasForm.tsx

```diff
@@ -0,0 +1,222 @@
+        ModelApiService.getModels
+    );
+
+    const modelUsageMap = useMemo(() => {
+        return (models ?? []).reduce((acc, model) => {
+            if (!acc.has(model.usage)) acc.set(model.usage, []);
+
+            acc.get(model.usage)?.push(model);
+
+            return acc;
+        }, new Map<string, Model[]>());
+    }, [models]);
+
+    const update = useAsync(
+        async (updates: UpdateDefaultModelAlias[]) => {
+            await Promise.all(
+                updates.map((update) =>
+                    DefaultModelAliasApiService.updateAlias(
+                        update.alias,
+                        update
+                    )
+                )
+            );
+        },
+        {
+            onSuccess: () => {
+                toast.success("Default model aliases updated");
+                onSuccess?.();
+            },
+        }
+    );
+
+    const defaultValues = useMemo(() => {
+        return defaultAliases?.reduce(
+            (acc, alias) => {
+                acc[alias.alias] = alias.model;
+                return acc;
+            },
+            {} as Record<string, string>
+        );
+    }, [defaultAliases]);
+
+    const form = useForm<Record<string, string>>({ defaultValues });
+
+    useEffect(() => {
+        return form.watch((values) => {
+            const changedItems = defaultAliases?.filter(({ alias, model }) => {
+                return values[alias] !== model;
+            });
+
+            if (!changedItems?.length) return;
+        }).unsubscribe;
+    }, [defaultAliases, form]);
+
+    useEffect(() => {
+        form.reset(defaultValues);
+    }, [defaultValues, form]);
+
+    const handleSubmit = form.handleSubmit((values) => {
+        const updates = defaultAliases
+            ?.filter(({ alias, model }) => values[alias] !== model)
+            .map(({ alias }) => ({
+                alias,
+                model: values[alias],
+            }));
+
+        update.execute(updates ?? []);
+    });
+
+    return (
+        <Form {...form}>
+            <form onSubmit={handleSubmit} className="space-y-6">
+                {defaultAliases?.map(({ alias, model: defaultModel }) => (
+                    <FormField
+                        control={form.control}
+                        name={alias}
+                        key={alias}
+                        render={({ field: { ref: _, ...field } }) => {
+                            const usage = getModelUsageFromAlias(alias);
+                            const modelOptions = usage
+                                ? modelUsageMap.get(usage)
+                                : [];
+
+                            return (
+                                <FormItem className="flex justify-between items-center space-y-0">
+                                    <FormLabel>{alias}</FormLabel>
+
+                                    <div className="flex flex-col gap-2 w-[50%]">
+                                        <FormControl>
```

Comment:
The code format looks weird in github..,

Reply:
There's a lot of tabs here... I'll clean this up a bit